### PR TITLE
Rename nodeupdate and nodeupdate_single and make them part of the official API

### PR DESCRIPTION
**Pull request by est31**
Now, the renamed forms of nodeupdate and nodeupdate_single are part of the official API.

As nodeupdate has been used by Minetest Game and in mods despite of not
being part of the official API, we ease the transition by still supporting
it for the 0.4.15 release. After the release, the two functions can be removed.

The removal will not violate the stability promise, as that promise only
includes the official and documented API.

Also, make some formerly global functions local. They most likely haven't
been used by mods, therefore they won't get stubs with deprecation warnings,
hard erroring directly.

See issue #4733 and PR #4753.

### CODE REVIEW

**Comment by Wuzzy2** [comment]
:+1: for only deprecating `nodeupdate` and `nodeupdate_single` while still adding new function names.
I would, however, write the deprecation explicitly into the changelog because this change is easy to miss.

File: builtin/game/falling.lua [hunk 1]
```diff
@@ -258,17 +258,41 @@ end
...
+	check_for_falling(p)
+end
+
+-- TODO remove this function after the 0.4.15 release
+function nodeupdate_single(p)
+	core.log("deprecated", "nodeupdate_single: deprecated, please use core.check_single_for_falling instead")
+	check_single_for_falling(p)
+end
+
+function core.check_for_falling(p)
```
Comment (paramat) [inline builtin/game/falling.lua]: Is doing this a Lua optimisation? If so should we be doing this everywhere from now on? and maybe edit builtin to do this throughout in another PR?

**Comment by paramat** [comment]
Line comment, but looks good.

File: builtin/game/falling.lua (hunk 1 above)
Comment (est31) [inline builtin/game/falling.lua]: @paramat I'm not sure anymore why I did this, but I agree it certainly was bad, its not really a speed enhancement.

File: builtin/game/falling.lua [hunk 2]
```diff
@@ -98,7 +98,7 @@ core.register_entity(":__builtin:falling_node", {
 				core.add_node(np, self.node)
 			end
 			self.object:remove()
-			nodeupdate(np)
+			check_for_falling(np)
```
Comment (rubenwardy) [inline builtin/game/falling.lua]: Shouldn't it be core.check_for_falling?

Comment (est31) [inline builtin/game/falling.lua]: Thanks for pointing out

File: doc/lua_api.txt [hunk 3]
```diff
@@ -2286,6 +2286,15 @@ and `minetest.auth_reload` call the authetification handler.
...
     * if `totallevel > maxlevel`, returns rest (`total-max`)
     * can be negative for decreasing
+* `core.check_single_for_falling(pos)`
+    * causes an unsupported `group:falling_node` node to fall and causes an
+      unattached `group:attached_node` node to fall.
+    * does not spread these updates to neighbours.
+* `core.check_for_falling(pos)`
+    * causes an unsupported `group:falling_node` node to fall and causes an
+      unattached `group:attached_node` node to fall.
+    * Spread these updates to neighbours and can cause a cascade
```
Comment (paramat) [inline doc/lua_api.txt]: Minor point, capital 'S' is inconsistent.

**Comment by paramat** [comment]
Minor line comment, however :+1:

**Comment by est31** [comment]
Adding own approval

**Comment by Zeno-** [comment]
:+1:

**Comment by paramat** [comment]
:+1:

**Comment by est31** [comment]
649448a2a91fbf3e944b2f2e739f4e2292af1df0
